### PR TITLE
[CPDLP-1945] Update NPQ guidance for outcomes and add a release note to API docs

### PR DESCRIPTION
**Pull request by leandroalemao**
### Context

We'll be adding new endpoints for NPQ outcomes into the API.. so we need to update the API guidance and release notes as well.

- Ticket: [CPDLP-1945](https://dfedigital.atlassian.net/browse/CPDLP-1945)

### Changes proposed in this pull request

Update NPQ guidance for outcomes and add a release note to the API docs according to the drafted material below:

[https://docs.google.com/document/d/1E-0sArwlJ64ZACwuLKI8M3dNdNru6uaMIZuK6cEW_EE/edit](https://docs.google.com/document/d/1E-0sArwlJ64ZACwuLKI8M3dNdNru6uaMIZuK6cEW_EE/edit)

[CPDLP-1910]: https://dfedigital.atlassian.net/browse/CPDLP-1910?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CPDLP-1945]: https://dfedigital.atlassian.net/browse/CPDLP-1945?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cwrw — COMMENTED** [review]
Looking good just a few comments mostly not introduced in this pr.

Also was wondering if we could separate NPQ declaration requests `NPQParticipantStartedDeclarationAttributes` to `NPQParticipantStartedDeclarationAttributes` and `NPQParticipantCompletedDeclarationAttributes` in v1 and v2, similar to v3? so we can add the has_passed to the latter as a required attribute.

Also in v3 I think we're missing the `/api/v1/participants/npq/outcomes`, could we add that there as well? Thanks

File: docs/source/_npq_usage_participant_actions.html.erb [hunk 1]
```diff
@@ -138,3 +138,70 @@
 <p class="govuk-body-m">
   For example, if a participant is on an npq-leadership-autumn schedule, the API will accept any type of declaration, such as a start, retention-1 or completion, from 1 November 2021.
 </p>
+
+<h2 id="declaring-that-an-npq-participant-has-completed-their-course" class="govuk-heading-l">Declaring that an NPQ participant has completed their course</h2>
```
[the remark this answers is not in the document]
Comment (leandroalemao) [reply docs/source/_npq_usage_participant_actions.html.erb:173]: Tks @cwrw for your review! It's been addressed in [96b7771](https://github.com/DFE-Digital/early-careers-framework/pull/2888/commits/96b777119e7b885b7ecd5494261227202c5f0ca1)

File: docs/source/_ecf_usage_participant_actions.html.erb [hunk 2]
```diff
@@ -324,3 +323,59 @@ takes precedence. In this instance, the provider should not change the mentor’
...
+<p class="govuk-body-m">See <a href="/api-reference/reference-v1#api-v1-participant-declarations-post" class="govuk-link">confirm ECF participant declarations</a> endpoint.</p>
+
+<h3 class="govuk-heading-m" id="provider-records-ECF-participant-declaration-id">2. Provider records the ECF participant declaration ID</h3>
+<p class="govuk-body-m">Store the returned ECF participant declaration ID for future management tasks.</p>
+
+<h2 id="declaring-that-an-ecf-participant-has-retained-their-course" class="govuk-heading-l">Declaring that an ECF participant has reached a retained milestone</h2>
+<p class="govuk-body-m">This scenario begins after it has been confirmed that an ECF participant has completed enough of their course to meet a milestone.</p>
+
+<h3 class="govuk-heading-m" id="provider-confirms-ECF-participant-has-been-retained">1. Provider confirms an ECF participant has been retained</h3>
+<p class="govuk-body-m">1. Confirm an ECF participant has been retained by the appropriate number of months for this retained event on their ECF course.</p>
```
Comment (peteryates) [inline docs/source/_ecf_usage_participant_actions.html.erb:347]: Looks like there are two `1.` prefixes here in a row.

```suggestion
<h3 class="govuk-heading-m" id="provider-confirms-ECF-participant-has-been-retained">1. Provider confirms an ECF participant has been retained</h3>
<p class="govuk-body-m">Confirm an ECF participant has been retained by the appropriate number of months for this retained event on their ECF course.</p>
```

**Review by peteryates — COMMENTED** [review]
I've had a scan through and it looks ok.

I wonder how necessary the numbered sections are though, the steps don't appear to be sequential and some of them only have one which looks a little odd. Probably more a content design question though cc @hazelyule

File: docs/source/_ecf_usage_participant_actions.html.erb (hunk 2 above)
Comment (leandroalemao) [reply docs/source/_ecf_usage_participant_actions.html.erb]: All fixed. Tks @peteryates

**Review by cwrw — APPROVED** [review]

**Pull request merged**